### PR TITLE
Pathwayz sidebar

### PR DESCRIPTION
**Pull request by ZeeJab**
Cleans up code from PR #247 for clearer understanding of what is happening:

More specifically it follows OOCSS conventions for cleaner files && a better understanding of what is going on for anyone that wants to debug later

Interesting fact: 
buttons files went from `160 lines to 98 o.O`
pathway-editor went from `45 lines to 26 -.o`
and html is much `cleaner` now

I was not able to refactor the js much but i did move the event handlers in editor.js on top of initialize as per backbone conventions. 

Once this PR is merged it adds the add, delete, undo, etc sidebar to the pathway pledge page as per #269

### CODE REVIEW

**Comment by stenington** [comment]
http://discovery-285.herokuapp.com/

**Comment by chloeatplay** [comment]
Quick q: why do the badge squares look so small? I am assuming they all look like they are on one column because @LucasBlair @ghami need to define positions, right?

**Comment by ZeeJab** [comment]
@chloeatplay I assumed it was because of `<div class="small-12 medium-8 medium-centered columns">` that wraps the canvas. This already exists in master so I did not remove it. @stenington?

**Comment by chloeatplay** [comment]
Hmm there are a bunch of things not quite looking as they should here
a) before pledging all the badges look tiny and in one straight column
discover open badges 2014-05-12 10-15-48

b) after pledging the badges look bigger but their title text doesn't wrap
discover open badges 2014-05-12 10-16-05
c) there are some funky things going on with the buttons, where their status remains blue after the interactions. In addition we will not need the "recommended" buttons for sure, and the "add notes" one is TBD i guess. 
d) the add badges section still does the dancing squares thingie, also those squares look huge @zzzndr whats the size for those? 
discover open badges 2014-05-12 10-16-33

**Comment by ZeeJab** [comment]
Okii, will look into it. But also keep in mind, the only thing that I've cleaned up in this PR is the actual sidebar and how it looks...nothing else has been worked on yet

**Comment by zzzndr** [comment]
for desktop the intended size for the badge containers is 210x210 pixels (:

**Comment by chloeatplay** [comment]
@ZeeJab okayz!

**Comment by stenington** [comment]
I'll fix the add panel in another PR, I have some mostly completed work that cleans all of that up that I can make a PR for after this one goes in.

**Comment by stenington** [comment]
That commit takes care of the pathway being overly narrow when not pledged.

File: clientapp/styles/add-badge-button.scss [hunk 1]
```diff
@@ -0,0 +1,37 @@
+.achievement .token {
```
Comment (stenington) [inline clientapp/styles/add-badge-button.scss]: This file seems weird given its name and contents. Any idea what happened @ZeeJab?

Comment (stenington) [inline clientapp/styles/add-badge-button.scss]: Oh, I guess it's styling achievements when they have a little checkbox for being added to pathways so it makes sense.

**Comment by stenington** [comment]
Styling of badges on a pathway is handled very differently than badges appearing anywhere else because it's done in canvas. Even though it looks wrong, we're probably better served by merging this once the buttons work and dealing with styling those separately.

**Comment by chloeatplay** [comment]
@stenington kk, makes sense :thumbsup:

File: clientapp/styles/add-badge-button.scss (hunk 1 above)
Comment (ZeeJab) [inline clientapp/styles/add-badge-button.scss]: yea :-( im cleaning this up now too

**Comment by stenington** [comment]
I pushed another version out to http://discovery-285.herokuapp.com/

**Comment by stenington** [comment]
Hmm, the "new" indicators are now way off the mark.

**Comment by stenington** [comment]
This PR has maybe gotten a little out of hand. I think I'm actually going to cherry-pick from it and issue many small PRs.

**Comment by ZeeJab** [comment]
@stenington we can close this now right?

**Comment by stenington** [comment]
yeppers

**Pull request closed without merging**